ank/src/cli_commands/server_connection.rs

```diff
@@ -219,13 +224,192 @@ impl ServerConnection {
+
+        self.listen_for_workload_logs(request_id, instance_names)
+            .await
+    }
+
+    async fn send_logs_request_for_workloads(
+        &mut self,
+        request_id: &str,
+        workload_instance_names: Vec<WorkloadInstanceName>,
+        args: LogsArgs,
+    ) -> Result<(), ServerConnectionError> {
+        let logs_request = LogsRequest {
+            workload_names: workload_instance_names,
+            follow: args.follow,
+            tail: args.tail,
+            since: args.since,
+            until: args.until,
+        };
+
+        self.to_server
+            .logs_request(request_id.to_string(), logs_request)
+            .await
+            .map_err(|err| ServerConnectionError::ExecutionError(err.to_string()))
+    }
+
+    async fn listen_for_workload_logs(
+        &mut self,
+        request_id: String,
+        mut instance_names: BTreeSet<WorkloadInstanceName>,
+    ) -> Result<(), ServerConnectionError> {
+        loop {
+            tokio::select! {
+                // [impl->swdd~cli-sends-logs-cancel-request-upon-termination~1]
+                _ = SignalHandler::wait_for_signals() => {
+                    self.to_server
+                        .logs_cancel_request(request_id).await
+                        .map_err(|err| ServerConnectionError::ExecutionError(err.to_string()))?;
+
+                    output_debug!("LogsCancelRequest sent after receiving signal to stop.");
+                    break Ok(());
+                }
+                // [impl->swdd~handles-log-responses-from-server~1]
+                server_message = self.from_server.recv() => {
+                    let server_message = server_message
+                        .ok_or(
+                            ServerConnectionError::ExecutionError("Error streaming workload logs: channel preliminary closed.".to_string()
+                    ))?;
+
+                    match response_to_log_streaming_state(&request_id, server_message)? {
+                        LogStreamingState::Output(log_entries) => {
+                            output_logs(log_entries.log_entries);
+                        }
+                        LogStreamingState::Continue => continue,
+                        // [impl->swdd~stops-log-output-for-specific-workloads~1]
+                        LogStreamingState::StopForWorkload(instance_name) => {
+                            instance_names.remove(&instance_name);
+
+                            if instance_names.is_empty() {
+                                // log streaming is finished for all requested instances
+                                output_debug!("All requested workload instances have been processed. Stopping log streaming.");
+                                break Ok(());
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+// [impl->swdd~handles-log-responses-from-server~1]
+fn response_to_log_streaming_state(
```

Comment:
The name is fitting to the functionality 👍, but I needed some time to get what it is supposed to do. 
What do you think about renaming to something close to the requirement id?
```suggestion
fn handle_server_log_response(
```

Reply:
Renamed in [0ffd626](https://github.com/eclipse-ankaios/ankaios/pull/499/commits/0ffd626ecf62bc5675a5c42c684878547f131611)